.circleci/local_publish_helpers.sh

```diff
@@ -12,6 +12,50 @@ function startLocalRegistry {
     grep -q 'http address' <(tail -f $tmp_registry_log)
 }
 
+function setNpmTag {
+    if [ -z $NPM_TAG ]; then
+        if [[ "$CIRCLE_BRANCH" =~ ^tagged-release ]]; then
+            if [[ "$CIRCLE_BRANCH" =~ ^tagged-release-without-e2e-tests\/.* ]]; then
+                export NPM_TAG="${CIRCLE_BRANCH/tagged-release-without-e2e-tests\//}"
+            elif [[ "$CIRCLE_BRANCH" =~ ^tagged-release\/.* ]]; then
+                export NPM_TAG="${CIRCLE_BRANCH/tagged-release\//}"
+            fi
+        fi
+        if [[ "$CIRCLE_BRANCH" == "beta" ]]; then
+            export NPM_TAG="beta"
+        fi
+    else
+        echo "NPM tag was already set!"
+    fi
+    echo $NPM_TAG
+}
+
+function uploadPkgCli {
+    cd out/
+    export hash=$(git rev-parse HEAD | cut -c 1-12)
+    export version=$(./amplify-pkg-linux --version)
+    sudo apt-get update
+    sudo apt-get install -y sudo tcl expect zip lsof jq groff python python-pip libpython-dev
+    sudo pip install awscli
+
+    aws s3 cp amplify-pkg-win.exe s3://pkg-cli-ci-do-not-delete/$(echo $version)/amplify-pkg-win-$(echo $hash).exe
+    aws s3 cp amplify-pkg-macos s3://pkg-cli-ci-do-not-delete/$(echo $version)/amplify-pkg-macos-$(echo $hash)
+    aws s3 cp amplify-pkg-linux s3://pkg-cli-ci-do-not-delete/$(echo $version)/amplify-pkg-linux-$(echo $hash)
```

Comment:
I'm a little confused about when these versions get used. Seems like this only runs during the deploy step which is only run on the release pipeline

Reply:
yes - they're only used in our e2e tests.